lib/sequelize.js

```diff
@@ -644,11 +644,11 @@ class Sequelize {
       const query = new this.dialect.Query(connection, this, options);
 
       try {
-        await this.runHooks('beforeQuery', options, query);
-        await checkTransaction();
+        await this.runHooks('beforeQuery', options, query, sql);
+        checkTransaction();
```

Comment:
```suggestion
        await checkTransaction();
```

just trying to see if the extra tick in between will cause the error to come back ...